app/auth/auth_service.ts

```diff
@@ -224,10 +224,17 @@ export class AuthService {
   }
 
   // Enters impersonation for the given group, which may either be a group ID or a URL identifier.
-  async enterImpersonationMode(query: string) {
+  async enterImpersonationMode(query: string, { redirectUrl = ""}: { redirectUrl?: string} = {}) {
     const request = grp.GetGroupRequest.create(query.startsWith("GR") ? { groupId: query } : { urlIdentifier: query });
     const response = await rpc_service.service.getGroup(request);
     this.setCookie(IMPERSONATING_GROUP_ID_COOKIE, response.id, { maxAge: 0 });
+
+    // If we have an explicit redirect URL, navigate there directly.
+    if (redirectUrl) {
+      window.location.href = redirectUrl
+      return
+    }
```

Comment:
does this play nicely with the subdomain logic below? (should redirectUrl be modified to have the correct subdomain?)

Reply:
The destination subdomain should only be different when "go to org" is used in which case redirectUrl would not be set.